node/src/main/scala/coop/rchain/node/runtime/Setup.scala

```diff
@@ -173,10 +180,19 @@ object Setup {
       blockProcessorQueue <- Queue.unbounded[F, (Casper[F], BlockMessage)]
       // block processing state - set of items currently in processing
       blockProcessorStateRef <- Ref.of(Set.empty[BlockHash])
+      unvalidated            <- blockStore.toMap
+      validated              <- blockMetadataStore.toMap
+      blockReceiverState <- BlockReceiver.create[F](
+                             blockRetriever,
+                             casperBufferStorage,
+                             unvalidated.keySet,
+                             validated.keySet
+                           )
```

Comment:
We cannot read the whole metadata store. It can take minutes .

Please supply functions from Dag storage which are necessary to check if block is already in the dag.

Reply:
Probably [done](https://github.com/rchain/rchain/pull/3650/commits/ecdff2c36e5773de6768b112b7542a066b85b4e7).